src/jit/codegenlegacy.cpp

```diff
@@ -18662,9 +18662,8 @@ regMaskTP CodeGen::genCodeForCall(GenTreeCall* call, bool valUsed)
 #ifdef _TARGET_ARM_
         if ((call->gtCallMoreFlags & GTF_CALL_M_SECURE_DELEGATE_INV))
         {
-            getEmitter()->emitIns_R_R_I(INS_add, EA_PTRSIZE, compiler->virtualStubParamInfo->GetReg(), regThis,
+            getEmitter()->emitIns_R_R_I(INS_add, EA_BYREF, compiler->virtualStubParamInfo->GetReg(), regThis,
                                         pInfo->offsetOfSecureDelegateIndirectCell);
-            regTracker.rsTrackRegTrash(compiler->virtualStubParamInfo->GetReg());
```

Comment:
This code is correct and shouldn't be moved.
The tracking logic keeps track of constants and LclVars that get loaded into registers, so that a second reload is issued.  

For example if at this point in time the register R4 contained the constant zero (from a prior instruction) after the INS_add instruction , we clear that info so that if we need a zero value we wouldn't use R4 after the the INS_add.

Reply:
(See lines 19279 to 19315) for a similar case:

What needs to be added is a rsLockReg/rsUnlockReg pair here and after the call instruction.